include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -144,16 +143,20 @@ void from_json(const BasicJsonType& j, std::forward_list<T, Allocator>& l)
 }
 
 // valarray doesn't have an insert method
-template<typename BasicJsonType, typename T,
-         enable_if_t<std::is_convertible<BasicJsonType, T>::value, int> = 0>
-void from_json(const BasicJsonType& j, std::valarray<T>& l)
+template<typename BasicJsonType, typename T>
+auto from_json(const BasicJsonType& j, std::valarray<T>& l) -> decltype(j.template get<T>(), void())
 {
     if (JSON_HEDLEY_UNLIKELY(!j.is_array()))
     {
         JSON_THROW(type_error::create(302, "type must be array, but is " + std::string(j.type_name())));
     }
     l.resize(j.size());
-    std::copy(j.begin(), j.end(), std::begin(l));
+    std::transform(
+        j.begin(), j.end(), std::begin(l),
```

Comment:
Can we use `std::begin` and `std::end` on `j` as well for consistency?

Reply:
I don't see the need to do that, since `j` is of type `json`, thus we know its member functions (c.f. `is_array` used a few lines above).

Also, I think this change is outside the scope of this PR.

Reply:
It just seems jarring to mix member and non-member in the same function call.  If we weren't already changing this line, then I wouldn't bring it up.  In any case, not something for which I would hold up the PR.